indexer/services/comlink/src/controllers/api/v4/affiliates-controller.ts

```diff
@@ -31,14 +38,49 @@ const controllerName: string = 'affiliates-controller';
 class AffiliatesController extends Controller {
   @Get('/metadata')
   async getMetadata(
-    @Query() address: string, // eslint-disable-line @typescript-eslint/no-unused-vars
+    @Query() address: string,
   ): Promise<AffiliateMetadataResponse> {
-    // simulate a delay
-    await new Promise((resolve) => setTimeout(resolve, 100));
+    // Check that the address exists
+    const walletRow = await WalletTable.findById(address);
+    if (!walletRow) {
+      throw new NotFoundError(`Wallet with address ${address} not found`);
+    }
+    const isVolumeEligible = Number(walletRow.totalVolume) >= config.VOLUME_ELIGIBILITY_THRESHOLD;
+
+    // Check if the address is an affiliate (has referred users)
+    const referredUserRows = await AffiliateReferredUsersTable.findByAffiliateAddress(address);
+    const isAffiliate = referredUserRows !== undefined ? referredUserRows.length > 0 : false;
+
+    // Get referral code (subaccount 0 username)
+    const subaccountRows = await SubaccountTable.findAll(
+      {
+        address,
+        subaccountNumber: 0,
+      },
+      [],
+    );
+    // No need to check subaccountRows.length > 1 as subaccountNumber is unique for an address
+    if (subaccountRows.length === 0) {
+      throw new UnexpectedServerError(`Subaccount 0 not found for address ${address}`);
+    }
+    const subaccountId = subaccountRows[0].id;
+
+    const usernameRows = await SubaccountUsernamesTable.findAll(
```

Comment:
instead of doing multiple awaits, can we do promise.All instead?

Reply:
Im not too famliar with ts, but the output of these functions is processed (for errors) and then used in subsequent functions. Would this be suitable for promise.all? My understanding is that promise.all should not be used for such cases?

Reply:
Linked the wrong line
but cant we combine 

const walletRow = await WalletTable.findById(address);
const referredUserRows = await AffiliateReferredUsersTable.findByAffiliateAddress(address);
    const subaccountRows = await SubaccountTable.findAll(
      {
        address,
        subaccountNumber: 0,
      },
      [],
    );
into one promise.all?
since all of these are independent and make db queries

Reply:
Can do!